### PR TITLE
BUGFIX: SD-692 The drop-down menu of Date picker is partially hidden

### PR DESCRIPTION
**Pull request by ptquang86**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [x] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2596
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2418

# Related Jira tasks
- SD-692: https://jira.intgdc.com/browse/SD-692

### CODE REVIEW

**Comment by ptquang86** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1253.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5313/ : SUCCESS in 11m 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/780/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/781/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2191/ : SUCCESS in 22m 56s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2762/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/782/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2192/ : UNSTABLE in 15m 42s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2763/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/783/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2193/ : SUCCESS in 38m 54s

File: src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx [hunk 1]
```diff
@@ -64,4 +64,42 @@ describe("ExtendedDateFilterBody", () => {
         const rendered = shallow(<DateFilterBody {...mockProps} isEditMode={false} />);
         expect(rendered.find(EditModeMessage)).not.toExist();
     });
+
+    describe("calculateHeight", () => {
+        const { innerWidth, innerHeight } = window;
```
Comment (naibinh) [inline src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx]: should we move this to `afterAll`?

File: src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx [hunk 2]
```diff
@@ -64,4 +64,42 @@ describe("ExtendedDateFilterBody", () => {
...
+
+        const getBodyWrapper = (component: ShallowWrapper) => {
+            return component.find(".gd-extended-date-filter-body-wrapper");
+        };
+
+        const getBodyScroller = (component: ShallowWrapper) => {
+            return component.find(".gd-extended-date-filter-body-scrollable");
+        };
+
+        afterAll(() => {
```
Comment (naibinh) [inline src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx]: Please help me check this `afterAll` is neccessary?
IMO, it's not.

File: src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx (hunk 1 above)
Comment (ptquang86) [reply src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx]: No, we should not. 
This should be run at the begining of the tests, to cache the size of the window.

File: src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx (hunk 2 above)
Comment (ptquang86) [reply src/components/filters/DateFilter/DateFilterBody/tests/DateFilterBody.spec.tsx]: removed.
thanks :)

File: src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx [hunk 3]
```diff
@@ -21,6 +21,11 @@ import { DateFilterHeader } from "./DateFilterHeader";
 import { DateFilterBodyButton } from "./DateFilterBodyButton";
 import { AbsolutePresetFilterItems } from "./AbsolutePresetFilterItems";
 
+const ACTIONS_BUTTONS_HEIGHT = 53;
+const EXCLUDE_OPEN_PERIOD_HEIGHT = 30;
```
Comment (naibinh) [inline src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx]: it should be `const EXCLUDE_OPEN_PERIOD_HEIGHT = 30; // height of 'Exclude open period' checkbox component`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5334/ : SUCCESS in 11m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/814/ : SUCCESS in 10s

**Comment by naibinh** [comment]
LGTM

File: src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx (hunk 3 above)
Comment (ptquang86) [reply src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx]: updated. thanks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5340/ : SUCCESS in 8m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/824/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5346/ : SUCCESS in 4m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/831/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2779/ : SUCCESS in 12m 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5347/ : SUCCESS in 12m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1297/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1775/ : SUCCESS in 19m 35s

**Pull request merged**